torch/_dynamo/eval_frame.py

```diff
@@ -643,6 +645,177 @@ def __bool__(self):
+
+    def placeholder(self, target, args, kwargs):
+        arg = super().placeholder(target, args, kwargs)
+        orig_inp_id = id(self.flat_args[self.count])
+
+        if orig_inp_id not in self.constraints_id_to_constraint:
+            self.count += 1
+            return arg
+
+        constraints = self.constraints_id_to_constraint[orig_inp_id]
+
+        # Convert simple sympy Integers into concrete int to
+        # insert into graph
+        def _convert_to_int(val):
+            if val == sympy.oo:
+                return math.inf
+            if isinstance(val, sympy.Integer):
+                return int(val)
+            # TODO ignore expressions for now
+            return None
+
+        for constraint in constraints:
+            constraint_range = constraint.constraint_range
+            if constraint_range is None:
+                continue
+
+            min_int_val = _convert_to_int(constraint_range.vr.lower)
+            max_int_val = _convert_to_int(constraint_range.vr.upper)
+
+            if min_int_val is None and max_int_val is None:
+                continue
+
+            dim = self.tracer.create_proxy(
+                "call_function", torch.ops.aten.sym_size, (arg, constraint.dim), {}
+            )
+            assert_msg = f"Input #{self.count}'s dimension #{constraint.dim} size is outside of supported dynamic range"
+
+            if min_int_val and min_int_val > 2:
+                gt = self.tracer.create_proxy(
+                    "call_function", operator.gt, (dim, min_int_val), {}
+                )
+                tensor_gt = self.tracer.create_proxy(
+                    "call_function", torch.scalar_tensor, (gt,), {}
+                )
+                self.tracer.create_proxy(
+                    "call_function",
+                    torch._assert_async,
+                    (tensor_gt, assert_msg),
+                    {},
+                )
+
+            if max_int_val and max_int_val < math.inf:
+                lt = self.tracer.create_proxy(
+                    "call_function", operator.lt, (dim, max_int_val), {}
+                )
+                tensor_lt = self.tracer.create_proxy(
+                    "call_function", torch.scalar_tensor, (lt,), {}
+                )
+                self.tracer.create_proxy(
+                    "call_function",
+                    torch._assert_async,
+                    (tensor_lt, assert_msg),
+                    {},
+                )
+
+        self.count += 1
+        return arg
+
+    def call_function(self, target, args, kwargs):
+        arg = super().call_function(target, args, kwargs)
+        from torch._export.constraints import constrain_as_size
+
+        if target == constrain_as_size:
+            min_val = kwargs["min"]
+            max_val = kwargs["max"]
+            gt = self.tracer.create_proxy(
+                "call_function", operator.gt, (args[0], min_val), {}
```

Comment:
Should we use `operator.ge` instead? I think the ranges are inclusive on both ends. (Same changes needed for many other places in this PR)

Reply:
Yep good catch. It should be `ge` and `le`